discovery/kubernetes/endpoints.go

```diff
@@ -453,15 +455,43 @@ func (e *Endpoints) buildEndpoints(eps *apiv1.Endpoints) *targetgroup.Group {
 	return tg
 }
 
+var objKeyPool = sync.Pool{}
+
+func generateObjKey(namespace, name string) (string, *strings.Builder) {
+	var sb *strings.Builder
```

Comment:
I'm afraid a builder would be an overkill here and we can do without its overhead, we just want to return `namespace/name` (endpoints are namespaced)
You can use `namespacedName` from `kubernetes.go`